plugins/SitesManager/Controller.php

```diff
@@ -229,6 +229,7 @@ public function siteWithoutDataTabs()
             'tagManagerActive' => $tagManagerActive,
             'consentManagerName' => false,
             'cloudflare' => $this->siteContentDetector->cloudflare,
+            'isCloud' => Manager::getInstance()->isPluginActivated('Cloud')
```

Comment:
@tsteur This check should be fine to proceed ?

Reply:
Personally I think we should try to avoid any code in core that is cloud only. If we need to replace some parts on cloud we can use e.g. events that allow a plugin to overwrite / replace certain parts. That way we can then replace e.g. the GTM intro within the cloud plugin.

And I guess at some point I would also start to refactor all that code again as having one big template that handles everything is a bit messy. Having e.g. separate classes for each type of detection might be smarter, easier to understand and better testable. Those classes could then have e.g. a method `renderTab` that allows providing additional tabs for the page (where needed) or a method to add sections to the `other ways` tab or similar. These methods could then have an event that allows to change/replace the content the method returns to make it adjustable on cloud.

Reply:
@sgiehl Removed `isCloud` and started using `postEvent` to override in other plugins

Reply:
@sgiehl Does this approach sounds good to you ?
Similarly can split it for Cloudflare and others in future